### ISSUE TITLE
Map.get with type variable keys

### ISSUE THREAD

**Issue by wmdietlGC** [typetools/checker-framework#1052]
This might have the same root cause as https://github.com/typetools/checker-framework/issues/1045 , but this time it's about the key in a Map.get being a type variable.

Take:

```
import java.util.Map;

class MapContains2 {
  String get1(Map<Object, String> map, Object k) {
    if (!map.containsKey(k)) {
      return "";
    }
    return map.get(k);
  }

  <KeyTV> String get2(Map<Object, String> map, KeyTV k) {
    if (!map.containsKey(k)) {
      return "";
    }
    return map.get(k);
  }
}
```

Running the Nullness Checker gives:

```
MapContains2.java:15: error: [return.type.incompatible] incompatible types in return.
    return map.get(k);
                  ^
  found   : @Initialized @Nullable String
  required: @Initialized @NonNull String
1 error
```

When the key is Object, as in get1, everything works as expected.

When the key is a type parameter, as in get2, I would expect the exact same behavior, but get the above error.

**Comment by smillst** [comment]
I believe this is a duplication of #961.

**Comment by smillst** [comment]
Closed in favor of #961.
Test case added here: checker/tests/nullness/Issue961.java
